src/odemis/driver/powerctrl.py

```diff
@@ -44,46 +46,62 @@ class PowerControlUnit(model.PowerSupplier):
     communication with the PCU firmware.
     '''
 
-    def __init__(self, name, role, port, pin_map=None, delay=None, init=None, ids=None,
+    def __init__(self, name, role, port, pin_map=None, delay=None, init=None,
+                 ids: Optional[List[str]] = None,
+                 excluded_ids: Optional[List[str]] = None,
                  termination=None, check_power=True, **kwargs):
-        '''
+        """
         port (str): port name
         pin_map (dict of str -> int): names of the components
           and the pin where the component is connected.
         delay (dict str -> float): time to wait for each component after it is
             turned on.
         init (dict str -> boolean): turn on/off the corresponding component upon
             initialization.
-        ids (list str): EEPROM ids expected to be detected during initialization.
+        ids (None or list[str]): EEPROM ids expected to be detected during initialization.
+        All the IDs must be found. If not, another device is search for.
+        excluded_ids (None or list[str]): EEPROM ids which must not be detected during initialization.
+        If any ID listed is detected, another device is search for, until a device with none
+        of these IDs is found.
         check_power (bool): normally the power is checked by verifying that the
           EEPROM IDs can be read. To disable this check, ids must be empty, and
           check_power set to False. This allows to use hardware which is not
           physically connected to EEPROMs.
         termination (dict str -> bool/None): indicate for every component
             if it should be turned off on termination (False), turned on (True)
             or left as-is (None).
-        Raise an exception if the device cannot be opened
-        '''
+        :raise:
+        * ValueError if an argument is invalid
+        * HwError if the device cannot be opened
+        """
+        model.PowerSupplier.__init__(self, name, role, **kwargs)
+
         if pin_map:
-            self.powered = list(pin_map.keys())
+            self._powered = list(pin_map.keys())
         else:
-            self.powered = []
-        model.PowerSupplier.__init__(self, name, role, **kwargs)
+            self._powered = []
 
         if not check_power and ids:
             raise ValueError(f"check_power can only be disabled if no ids are specified, but got {ids}")
 
         self._check_power = check_power
 
-        # TODO: catch errors and convert to HwError
+        required_ids = ids or []
+        if not all(isinstance(eid, str) and len(eid) >= 8 for eid in required_ids):
+            raise ValueError(f"ids must be a list of strings (of at least 8 chars), but got '{ids}'")
+
+        excluded_ids = excluded_ids or []
+        if not all(isinstance(eid, str) and len(eid) >= 8 for eid in excluded_ids):
+            raise ValueError(f"excluded_ids must be a list of strings (of at least 8 chars), but got '{excluded_ids}'")
```

Comment:
Does it make sense to mention in the docstring that the ids and excluded_ids must be at least 8 chars?

Reply:
All EEPROM IDs are above 8 chars. So it'd be redundant. I've just added this check to detect typos in the arguments.